### DISCUSSION TITLE
Copilot suggested API Key as well

### DISCUSSION THREAD

**Discussion by swadhinbharat** [community/community#21267] [Copilot Conversations]
I was defining a URL in my GO app and Copilot completed my URL with an API key as well for a certain site.

**Comment by davecheney** [comment]
Was is the correct key?

**Reply by swadhinbharat** [reply]
Yes. It worked fine upon testing.

**Comment by MrSarno** [comment]
Unfortunately, this has been a persistent issue since Copilot's inception;
https://fossbytes.com/github-copilot-generating-functional-api-keys/

Additionally, it appears to enjoy pilfering people's code and then lying about the license;
https://twitter.com/mitsuhiko/status/1410886329924194309

The year-old tweet linked-to above also shows it doesn't appear to discard any comments featuring profanity.

Copilot is simultaneously highly impressive and staggeringly primitive. I don't think it is ready for release now, let alone a year ago.

_(**Note:** Please don't take my comment the wrong way; it's good that people continue reporting this so GitHub don't forget about it.)_

**Reply by derekxmartin** [reply]
Would this happen if one were to use OpenAI Codex? Hard to know who is at-fault here re the keys. I suspect that an additional filter added onto the HTTP response that removes credentials (think pentest sniffing tool-like functionality) would address this.

GPT-3 is guaranteed to be interesting but never inherently correct. I would love to hear thoughts pro/const.

**Reply by MrSarno** [reply]
@derekmartinjs I am _extremely_ confident when I say you know vastly more about this than I do. 😃

Maybe someone else will weigh in with their thoughts.
